server/src/main/java/org/elasticsearch/index/mapper/vectors/MultiDenseVectorFieldMapper.java

```diff
@@ -0,0 +1,443 @@
+                "Field ["
+                    + fullPath()
+                    + "] of type ["
+                    + typeName()
+                    + "] doesn't support indexing multiple values for the same field in the same document"
+            );
+        }
+        if (XContentParser.Token.VALUE_NULL == context.parser().currentToken()) {
+            return;
+        }
+        if (XContentParser.Token.START_ARRAY != context.parser().currentToken()) {
+            throw new IllegalArgumentException(
+                "Field [" + fullPath() + "] of type [" + typeName() + "] cannot be indexed with a single value"
+            );
+        }
+        if (fieldType().dims == null) {
+            int currentDims = -1;
+            while (XContentParser.Token.END_ARRAY != context.parser().nextToken()) {
+                int dims = fieldType().elementType.parseDimensionCount(context);
+                if (currentDims == -1) {
+                    currentDims = dims;
+                } else if (currentDims != dims) {
+                    throw new IllegalArgumentException(
+                        "Field [" + fullPath() + "] of type [" + typeName() + "] cannot be indexed with vectors of different dimensions"
+                    );
+                }
+            }
+            MultiDenseVectorFieldType updatedFieldType = new MultiDenseVectorFieldType(
+                fieldType().name(),
+                fieldType().elementType,
+                currentDims,
+                indexCreatedVersion,
+                fieldType().meta()
+            );
+            Mapper update = new MultiDenseVectorFieldMapper(leafName(), updatedFieldType, builderParams, indexCreatedVersion);
+            context.addDynamicMapper(update);
+            return;
+        }
+        int dims = fieldType().dims;
+        DenseVectorFieldMapper.ElementType elementType = fieldType().elementType;
+        List<VectorData> vectors = new ArrayList<>();
+        while (XContentParser.Token.END_ARRAY != context.parser().nextToken()) {
+            VectorData vector = elementType.parseKnnVector(context, dims, (i, b) -> {
+                if (b) {
+                    checkDimensionMatches(i, context);
+                } else {
+                    checkDimensionExceeded(i, context);
+                }
+            }, null);
+            vectors.add(vector);
+        }
+        int bufferSize = elementType.getNumBytes(dims) * vectors.size();
+        ByteBuffer buffer = ByteBuffer.allocate(bufferSize).order(ByteOrder.LITTLE_ENDIAN);
+        ByteBuffer magnitudeBuffer = ByteBuffer.allocate(vectors.size() * Float.BYTES).order(ByteOrder.LITTLE_ENDIAN);
+        for (VectorData vector : vectors) {
+            vector.addToBuffer(buffer);
+            magnitudeBuffer.putFloat((float) Math.sqrt(elementType.computeSquaredMagnitude(vector)));
+        }
+        String vectorFieldName = fieldType().name();
+        String vectorCountFieldName = vectorFieldName + VECTOR_COUNT_SUFFIX;
```

Comment:
Do we really need to create an extra field for that?  Would it better to calculate counts dynamically based on number of dims and byteref length?

Reply:
Ah yes! We could do that!